viewport/test/with-viewport-match.js

```diff
@@ -15,14 +16,28 @@ import '../store';
 import withViewportMatch from '../with-viewport-match';
 
 describe( 'withViewportMatch()', () => {
-	const Component = () => <div>Hello</div>;
+	const ChildComponent = () => <div>Hello</div>;
+
+	// this is needed because TestUtils does not accept a stateless component.
+	// anything run through a HOC ends up as a stateless component.
+	const getTestComponent = ( WrappedComponent ) => {
+		class TestComponent extends Component {
+			render() {
+				return <WrappedComponent { ...this.props } />;
+			}
+		}
+		return <TestComponent />;
+	};
 
 	it( 'should render with result of query as custom prop name', () => {
 		dispatch( 'core/viewport' ).setIsMatching( { '> wide': true } );
-		const EnhancedComponent = withViewportMatch( { isWide: '> wide' } )( Component );
-		const wrapper = mount( <EnhancedComponent /> );
+		const EnhancedComponent = withViewportMatch(
+			{ isWide: '> wide' }
+		)( ChildComponent );
+		const wrapper = renderer.create( getTestComponent( EnhancedComponent ) );
 
-		expect( wrapper.find( Component ).props() ).toEqual( { isWide: true } );
+		expect( wrapper.root.findByType( ChildComponent ).props.isWide )
```

Comment:
It isn't exactly the same check. Can we update it to make sure it doesn't change anything?

Reply:
I'm sorry I'm not sure exactly what you mean by "Can we update it to make sure it doesn't change anything?".  I'm also not sure why you think this isn't the same check.  I had to wrap `ChildComponent` in the `TestComponent` so effectively I'm doing the same test, just doing a different method for getting the component to test.  This same method will apply when HOCs are returning a forwardRef wrapped component (in 7557).

Reply:
Another alternative here would be to use `TestUtils` to do a full mount into the DOM and then traverse actual HTMLElement nodes to verify expected results?  But I'm not sure that is any better than what is happening here?

Reply:
```js
expect( wrapper.root.findByType( ChildComponent ).props ).toEqual( { isWide: true });
```
This is what I thought about. Does `ref` causes any issues here?

Reply:
I ensured these tests passed against 7557 (while preserving the intent of the test).  Then ensured the same tests passed against master (if they didn't pass against master then I knew there was a problem).

Reply:
It's minor, we can skip it :)